### PR TITLE
Better experience when no actions are defined

### LINKED ISSUE #4440: "Trends" tab layout seems broken
## Bug description

The "Insights" tab seems to run in a single-column layout as opposed to its original two-column layout. Guessing some layout was mistakenly tweaked. 

`Insights` page > `Trends` tab right now:

CleanShot 2021-05-21 at 16 06 37@2x

## Expected behavior

`Funnels` tab right now - this is how Insights tab should also look like

CleanShot 2021-05-21 at 16 06 48@2x

## How to reproduce

1. Go to `Insights` page > `Trends` tab

## Environment

- [x] PostHog Cloud
- [ ] self-hosted PostHog - Didn't check

### PR DESCRIPTION
**Pull request by samwinslow**
## Changes

UPDATE:

Drop the plain copy, add the following rich message anywhere ActionFilterDropdown is used and actions are empty. Clicking the list item or the button will take the user to `/actions`.

Screen Shot 2021-05-25 at 11 35 24 AM

This PR also:

- adds support for customizing `onSelect` behavior in the SelectBox on a per-item basis
- gives the option to prevent the default select handler from running, if desired
- adds support for overriding the Info panel on the right side of SelectBox on a per-item basis

----

OLD:

Fixes #4440 (the more specific request was resolved via email). The "missing actions" microcopy looked strange and an extra linebreak should fix it.

Screen Shot 2021-05-24 at 9 11 17 AM

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests
- [ ] Migrations are safe to run at scale (e.g. PostHog Cloud) – present proof if not obvious

### CODE REVIEW

**Review by paolodamico — COMMENTED** [review]
Actually I think it'd be better to drop this CTA altogether and instead add it in the _Actions_ section of the filter row picker (also recommend adding some small graphic and having the CTA muted as not to draw too much attention). We've now seen several cases of users just using raw events (mainly cause they don't use autocapture), so not having actions is fine. 

Add CTA in this section

File: frontend/src/scenes/insights/ActionFilter/ActionFilterRow/ActionFilterDropdown.tsx [hunk 1]
```diff
@@ -114,13 +103,48 @@ export function ActionFilterDropdown({
...
+                        ? actions.map((action: ActionType) => ({
+                              key: EntityTypes.ACTIONS + action.id,
+                              name: action.name,
+                              volume: action.count,
+                              id: action.id,
+                              action,
+                          }))
+                        : [
+                              {
+                                  key: EntityTypes.ACTIONS + 'create_new',
```
Comment (paolodamico) [inline frontend/src/scenes/insights/ActionFilter/ActionFilterRow/ActionFilterDropdown.tsx:116]: Nit: `EntityTypes.ACTIONS + '_create_new'`

File: frontend/src/scenes/insights/ActionFilter/ActionFilterRow/ActionFilterDropdown.tsx [hunk 2]
```diff
@@ -183,7 +175,10 @@ export function ActionFilterDropdown({
     )
 }
 
-export function ActionInfo({ item }: { item: SelectedItem }): JSX.Element {
+export function ActionInfo({ item }: RenderInfoProps): JSX.Element {
```
Comment (paolodamico) [inline frontend/src/scenes/insights/ActionFilter/ActionFilterRow/ActionFilterDropdown.tsx:178]: Nit: I don't think these need to be exported.

**Review by paolodamico — APPROVED** [review]
🚢🚢🚢

File: frontend/src/scenes/insights/ActionFilter/ActionFilterRow/ActionFilterDropdown.tsx (hunk 2 above)
Comment (samwinslow) [reply frontend/src/scenes/insights/ActionFilter/ActionFilterRow/ActionFilterDropdown.tsx:178]: `ActionInfo` is used in `SessionsFilterBox` for sessions filtering, but you're correct that the others don't need to be exported.

**Pull request merged**